### PR TITLE
Adds /refreshrs/teachers/:teacherId endpoint

### PR DESCRIPTION
**Pull request by dividedsky**
For fetching all teacher's refreshrs on the followup list page

### CODE REVIEW

File: server/routes/refreshrsRoutes.js [hunk 1]
```diff
@@ -14,6 +14,24 @@ router.get('/', async (req, res, next) => {
...
+router.get('/teachers/:teacherId', async (req, res, next) => {
+  try {
+    const { teacherId } = req.params;
+    const refreshrList = await db.getTeacherRefreshrs(teacherId);
+    console.log(refreshrList);
+    res.status(responseStatus.success).json(refreshrList);
+  } catch (err) {
+    console.log(err);
+    if (TypeError) {
+      next(responseStatus.notFound);
```
Comment (nickoferrall) [inline server/routes/refreshrsRoutes.js]: Interesting - so `if (TypeError)` means if the information is the wrong data / type of data, log notFound error?

File: server/routes/refreshrsRoutes.js [hunk 2]
```diff
@@ -14,6 +14,24 @@ router.get('/', async (req, res, next) => {
...
+    } else {
+      next(err);
```
Comment (nickoferrall) [inline server/routes/refreshrsRoutes.js]: My understanding of `next` is that it signifies the end of a middleware function. What does it do here? What would happen without it?

File: server/routes/refreshrsRoutes.js (hunk 1 above)
Comment (kelfro) [reply server/routes/refreshrsRoutes.js]: Yes, it looks like there is going to be a notfound error if the information is of the wrong type (integer, string, any).

File: server/routes/refreshrsRoutes.js (hunk 2 above)
Comment (kelfro) [reply server/routes/refreshrsRoutes.js]: https://expressjs.com/en/guide/error-handling.html
I think with an async request, you need to use next to make sure Express processes the error.

Comment (dividedsky) [reply server/routes/refreshrsRoutes.js]: These are good questions that I don't have an answer to. :smile: I was kinda copying the catches in the other endpoints. I think next would pass it to the error handler? We can refine it later, for sure.

Comment (dividedsky) [reply server/routes/refreshrsRoutes.js]: Cool, thanks for that link, Kelly!

**Pull request merged**